test/auth/oauth.tests.js

```diff
@@ -465,4 +467,175 @@ describe('OAuthAuthenticator', function () {
     });
   });
 
+  describe('#clientCredentials', function () {
+    var path = '/oauth/token';
+    var options = {
+      audience: 'audience',
+      scope: 'scope'
+    };
+
+    beforeEach(function () {
+      this.authenticator = new Authenticator(validOptions);
+      this.request = nock(API_URL)
+        .post(path)
+        .reply(200);
+    });
+
+
+    it('should require an object as first argument', function () {
+      expect(this.authenticator.clientCredentialsGrant)
+        .to.throw(ArgumentError, 'Missing options object');
+    });
+
+    it('should accept a callback', function (done) {
+      this
+        .authenticator
+        .clientCredentialsGrant(options, done.bind(null, null));
+    });
+
+
+    it('should return a promise when no callback is provided', function (done) {
+      this
+        .authenticator
+        .clientCredentialsGrant(options)
+        .then(done.bind(null, null))
+        .catch(done.bind(null, null));
+    });
+
+
+    it('should perform a POST request to ' + path, function (done) {
+      var request = this.request;
+
+      this
```

Comment:
you can return the promise and avoid calling `done()` inside `then`.

I would not add `catch(done)`. If this results in an error, the test should not pass